go-controller/pkg/ovn/endpoints.go

```diff
@@ -108,12 +108,24 @@ func (ovn *Controller) AddEndpoints(ep *kapi.Endpoints) error {
 				}
 			}
 			// Cloud load balancers: directly load balance that traffic from pods
+			// Apply to gateway load-balancers to handle ingress traffic to the GR as well as worker switches
 			for _, ing := range svc.Status.LoadBalancer.Ingress {
 				if ing.IP == "" {
```

Comment:
I'm not sure. This is how it was in the code before so I just kept it. What documentation are you looking at?

Reply:
Sorry, I was looking at the doc bloc for `LoadBalancerIngress`:
```go
// LoadBalancerIngress represents the status of a load-balancer ingress point:
// traffic intended for the service should be sent to an ingress point.
type LoadBalancerIngress struct {
	// IP is set for load-balancer ingress points that are IP based
	// (typically GCE or OpenStack load-balancers)
	// +optional
	IP string `json:"ip,omitempty" protobuf:"bytes,1,opt,name=ip"`

	// Hostname is set for load-balancer ingress points that are DNS based
	// (typically AWS load-balancers)
	// +optional
	Hostname string `json:"hostname,omitempty" protobuf:"bytes,2,opt,name=hostname"`
}
```

Reply:
Ah I see. I'm not sure how we would handle a hostname, since we don't offer any external DNS to the cluster. I guess we would have to do the DNS lookup ourselves and then use the returning IPs as the VIPs. The comment says AWS load-balancers which handles DNAT of the IP to the cluster. I would suggest we just say we do not support handling Ingress packets that do not have the IP option set.

Reply:
OK, I don't know enough about this to evaluate its importance. I was thinking we could use the same DNS lookup egress firewall is using? For future work of course.

Reply:
I couldn't see any place in kube-proxy that handles the Hostname field, I would not consider it

Reply:
@alexanderConstantinescu  please create a jira issue for this and we can track it there, thanks.

Reply:
Created SDN-1368